Scripts/VillagerVariety.cs

```diff
@@ -0,0 +1,173 @@
+using System.Reflection;
+
+namespace VillagerVariety
+{
+    public class VillagerVariety : MonoBehaviour
+    {
+        #region Fields
+
+        public static Mod mod;
+
+        public const int NUM_VARIANTS = 2;  // Number of variants to generate, a variant falls back to 0 if no images found.
+        public readonly static string[] seasonStrs = { "f", "p", "m", "w" };
+
+        // Face data from MobilePersonNPC
+        static int[] maleRedguardFaceRecordIndex = new int[] { 336, 312, 336, 312 };
+        static int[] femaleRedguardFaceRecordIndex = new int[] { 144, 144, 120, 96 };
+        static int[] maleNordFaceRecordIndex = new int[] { 240, 264, 168, 192 };
+        static int[] femaleNordFaceRecordIndex = new int[] { 72, 0, 48, 0 };
+        static int[] maleBretonFaceRecordIndex = new int[] { 192, 216, 288, 240 };
+        static int[] femaleBretonFaceRecordIndex = new int[] { 72, 72, 24, 72 };
+
+        static VillagerVariety instance;
+
+        #endregion
+
+        #region Public Methods
+
+        [Invoke(StateManager.StateTypes.Start, 0)]
+        public static void Init(InitParams initParams)
+        {
+            mod = initParams.Mod;
+            instance = new GameObject(mod.Title).AddComponent<VillagerVariety>();
+
+            mod.MessageReceiver = MessageReceiver;
+
+            // Register events
+            PlayerEnterExit.OnTransitionDungeonExterior += OnTransition;
+            PlayerEnterExit.OnTransitionExterior += OnTransition;
+            PlayerGPS.OnEnterLocationRect += OnTransition;
+
+            mod.IsReady = true;
+        }
+
+        public static string GetImageName(int archive, int record, int frame, int face, int variant, string season)
+        {
+            return string.Format("{0:000}.{3}.{4}{5}_{1}-{2}", archive, record, frame, face, variant, season);
+        }
+        public static string GetImageName(int archive, int record, int frame, int variant, string season)
+        {
+            return string.Format("{0:000}.{3}.{4}{5}_{1}-{2}", archive, record, frame, "X", variant, season);
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        static void OnTransition(PlayerEnterExit.TransitionEventArgs args)
+        {
+            OnTransition();
+        }
+        static void OnTransition(DaggerfallConnect.DFLocation location)
+        {
+            OnTransition();
+        }
+        static void OnTransition()
+        {
+            instance.StartCoroutine(PreloadPopulation());
+        }
+
+        static IEnumerator PreloadPopulation()
+        {
+            // If in town
+            PopulationManager pm = GameObject.FindObjectOfType<PopulationManager>();
+            if (pm == null) yield break;
+
+            // Wait for PopulationManager to call Start
+            yield return null;
+
+            // Ready billboard
+            VillagerVarietyMobilePerson billboard = new GameObject().AddComponent<VillagerVarietyMobilePerson>();

```

Comment:
Seems a bit wasteful to create a new gameobject and a new `VillagerVarietyMobilePerson` on each variation just to cache the textures. Perhaps this could be refactored to only do the texture generation/caching part?